daemon/logger/ring.go

```diff
@@ -0,0 +1,219 @@
+package logger
+
+import (
+	"errors"
+	"sync"
+
+	"github.com/Sirupsen/logrus"
+	"golang.org/x/net/context"
+)
+
+const (
+	defaultRingMaxSize = 1000000 // 1MB
+)
+
+// RingLogger is a ring buffer that implements the Logger interface.
+// This is used when lossy logging is OK.
+type RingLogger struct {
+	buffer *messageRing
+	ctx    context.Context
```

Comment:
That's true, it's nice on the for { select }, but since we're only selecting to check for closed I guess it makes sense.